annotate.el

```diff
@@ -1567,36 +1574,39 @@ The searched interval can be customized setting the variable:
                        (when force-newline-p
                          (annotate-annotation-force-newline-policy annotation))
                        annotation))))
-      (if (not (annotate-string-empty-p annotated-text))
-          (let ((text-to-match (ignore-errors
-                                 (buffer-substring-no-properties start end))))
-            (if (and text-to-match
-                     (string= text-to-match annotated-text))
-                (create-annotation start end annotation-text)
-              (let* ((starting-point-matching (go-backward start))
-                     (ending-point-match      (go-forward  start))
-                     (length-match            (- end start))
-                     (new-match               (guess-match-and-add starting-point-matching
-                                                                   (+ starting-point-matching
-                                                                      length-match)
-                                                                   annotated-text
-                                                                   ending-point-match)))
-                (and new-match
-                     (create-annotation new-match
-                                        (+ new-match length-match)
-                                        annotation-text)))
-              (lwarn '(annotate-mode) ; if matches annotated text failed
-                     :warning
-                     annotate-warn-file-searching-annotation-failed-control-string
-                     (annotate-actual-file-name)
-                     annotation-text
-                     text-to-match)))
-        (create-annotation start end annotation-text)) ; create new annotation
-      (when (use-region-p)
-        (deactivate-mark))
-      (save-excursion
-        (goto-char end)
-        (font-lock-fontify-block 1))))
+    (if (annotate-string-empty-p annotation-text)
+        (signal 'annotate-empty-annotation-text-error t)
+      (progn
+        (if (not (annotate-string-empty-p annotated-text))
```

Comment:
I think we are already in the `else` clause here, and can skip the `(if (not ... ))`.